### PR TITLE
Refactor code name sanitization; simplify

### LINKED ISSUE #265: Datavyu Scripting NoMethoderror
Hello!
I'm trying to learn how to code in datavyu and I keep getting the following error no matter what I do.
It's just a simple script to copy the data from one column to another.

*************************
Running Script: CopyColumn1.rb on project: datavyu001
*************************
NoMethodError: undefined method `chr' for nil:NilClass
   set_cells at filepath/Datavyu_API.rb:360
        each at org/jruby/RubyArray.java:1613
   set_cells at filepath/Datavyu_API.rb:357
  get_column at filepath/Datavyu_API.rb:751
      (root) at <script>:4

***** SCRIPT ERROR *****

NoMethodError: undefined method `chr' for nil:NilClass
   set_cells at filepath/Datavyu_API.rb:360
        each at org/jruby/RubyArray.java:1613
   set_cells at filepath/Datavyu_API.rb:357
  get_column at filepath/Datavyu_API.rb:751
      (root) at <script>:4

See line 4 of filepath\CopyColumn1.rb:
	fill = get_column("fill");
*************************

I've only got a bit of experience in c++ coding but I can't get past this error message.

Any help would be greatly appreciated!

W

### PR DESCRIPTION
**Pull request by sh0sh1n**
Closes #265 #266

### CODE REVIEW

File: src/main/resources/Datavyu_API.rb [hunk 1]
```diff
@@ -194,14 +193,17 @@ def change_code(arg, val)
     elsif arg == "ordinal"
       val = val.to_i if val.class == String
       @ordinal = val
-    elsif @arglist.include?(arg)
-      for i in 0..arglist.length-1
-        if arglist[i] == arg and not arg.nil?
-          argvals[i] = val.to_s
+    else
+      san_arg = RColumn.sanitize_codename(arg)
```
Comment (TheoWolf) [inline src/main/resources/Datavyu_API.rb]: Why not using convert_argname(arg) function?

File: src/main/resources/Datavyu_API.rb [hunk 2]
```diff
@@ -432,15 +433,14 @@ def sort_cells()
   def change_code_name(old_name, new_name)
     i = @old_args.index(old_name)
     @old_args[i] = new_name
-    if ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"].include?(old_name[1].chr)
```
Comment (TheoWolf) [inline src/main/resources/Datavyu_API.rb]: Does the RColumn.sanitize.codename function will check this condition and process the arg accordingly, or it is obsolete and could be removed?

File: src/main/resources/Datavyu_API.rb [hunk 3]
```diff
@@ -452,14 +452,11 @@ def change_code_name(old_name, new_name)
   # @return nil
   def add_code(name)
     @old_args << name
-    if ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"].include?(name[1].chr)
```
Comment (TheoWolf) [inline src/main/resources/Datavyu_API.rb]: See comment above.

File: src/main/resources/Datavyu_API.rb [hunk 4]
```diff
@@ -2256,10 +2251,7 @@ def transfer_columns(db1, db2, remove, *varnames)
 #   check_reliability("trial", "rel.trial", "trialnum", 100)
 def check_reliability(main_col, rel_col, match_arg, time_tolerance, *dump_file)
   # Make the match_arg conform to the method format that is used
-  if ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"].include?(match_arg[0].chr)
```
Comment (TheoWolf) [inline src/main/resources/Datavyu_API.rb]: See comment above

File: src/main/resources/Datavyu_API.rb [hunk 5]
```diff
@@ -2490,10 +2479,7 @@ def check_valid_codes2(data, outfile, *arg_filt_pairs)
     		end
 
     		arg = arg_filt_pairs[i]
-    		if ["0","1","2","3","4","5","6","7","8","9"].include?(arg[1].chr)
```
Comment (TheoWolf) [inline src/main/resources/Datavyu_API.rb]: See comment above

**Comment by TheoWolf** [comment]
@sh0sh1n Do you feel confident adding this branch to 1.5.0 release or maybe you would like to do more testing?

**Comment by sh0sh1n** [comment]
Ok, I will review this today. This has been on the back burner but I'd like to have these changes go into a major release rather than a minor one.

**Comment by sh0sh1n** [comment]
Note the force push to bring this up to date with dev branch and keep the timeline clean.

**Pull request merged**